examples/http/hpe_msa_san_disks.js

```diff
@@ -0,0 +1,186 @@
+ * 
+ */
+
+// Variable to store the session key
+var sessionKey;
+
+// Create a Custom Driver table to store disks data
+var table = D.createTable(
+    "Disks",
+    [
+        { label: "Serial Number", valueType: D.valueType.STRING },
+        { label: "Usage", valueType: D.valueType.STRING },
+        { label: "Disk Group", valueType: D.valueType.STRING },
+        { label: "Pool", valueType: D.valueType.STRING },
+        { label: "Tier", valueType: D.valueType.STRING },
+        { label: "Status", valueType: D.valueType.STRING },
+        { label: "Health", valueType: D.valueType.STRING },
+    ]
+);
+
+// Process the response from the server
+function processResponse(d) {
+    return function process(error, response, body) {
+        if (error) {          
+            console.error(error);
+            D.failure(D.errorType.GENERIC_ERROR);
+        } else if (response.statusCode == 404) {
+            D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+        } else if (response.statusCode == 401 || response.statusCode == 403) {
+            D.failure(D.errorType.AUTHENTICATION_ERROR);
+        } else if (response.statusCode != 200) {
+            D.failure(D.errorType.GENERIC_ERROR);
+        } 
+        if (response.headers["command-status"]) {
+            sessionKey = response.headers["command-status"].split(/^.*?\s/)[1];
+            if(sessionKey == "Authentication Unsuccessful"){
+                console.error("Session key not found in response headers");
+                D.failure(D.errorType.AUTHENTICATION_ERROR);
+            }
+        }
+        d.resolve(body);      
+    };
+}
+
+/**
+ * Logs in to the HPE MSA SAN device using basic authentication
+ * @returns A promise that resolves on successful login
+ */
+function login() {
+    var d = D.q.defer();
+    var config = {
+        url: "/api/login",
+        username: D.device.username(),
+        password: D.device.password(),
+        protocol: "https",
+        auth: "basic",
+        jar: true,
+        rejectUnauthorized: false
+    };
+    D.device.http.get(config, processResponse(d));
+    return d.promise;
+}
+
+// Retrieves disks data from the HPE MSA SAN device
+function getDisks() {
+    var d = D.q.defer();
+    var config = {
+        url: "/api/show/disks",
+        protocol: "https",
+        jar: true,
+        rejectUnauthorized: false,
+        headers: {
+            "sessionKey": sessionKey 
+        }
+    };
+    D.device.http.get(config, processResponse(d));
+    return d.promise;
+}
+
+// Sanitize location value to create a recordId
+function sanitize(output){
+    var recordIdReservedWords = ['\\?', '\\*', '\\%', 'table', 'column', 'history'];
+    var recordIdSanitisationRegex = new RegExp(recordIdReservedWords.join('|'), 'g');
+    return output.replace(recordIdSanitisationRegex, '').slice(0, 50).replace(/\s+/g, '-').toLowerCase();
+}
+
+// Extracts relevant data from the API response
+function extractData(data) {
```

Comment:
use the xml library instead of manual parsing
D.htmlParse(xml)
refer to some examples done before like aws_ec2_volumes.js